server/test/list_objects.go

```diff
@@ -203,33 +338,149 @@ func runListObjectsTests(t *testing.T, ctx context.Context, testCases []listObje
 	}
 }
 
-func setupTestListObjects(store string, datastore storage.OpenFGADatastore) (context.Context, storage.OpenFGADatastore, string, error) {
+// Used to avoid compiler optimizations (see https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go)
+var listObjectsResponse *openfgapb.ListObjectsResponse //nolint
+
+func BenchmarkListObjectsWithReverseExpand(b *testing.B, ds storage.OpenFGADatastore) {
+
 	ctx := context.Background()
-	data, err := os.ReadFile(gitHubTestDataFile)
-	if err != nil {
-		return nil, nil, "", err
+	store := ulid.Make().String()
+
+	model := &openfgapb.AuthorizationModel{
+		Id:            ulid.Make().String(),
+		SchemaVersion: typesystem.SchemaVersion1_0,
+		TypeDefinitions: []*openfgapb.TypeDefinition{
+			{
+				Type: "user",
+			},
+			{
+				Type: "document",
+				Relations: map[string]*openfgapb.Userset{
+					"viewer": {
+						Userset: &openfgapb.Userset_This{},
+					},
+				},
+				Metadata: &openfgapb.Metadata{
+					Relations: map[string]*openfgapb.RelationMetadata{
+						"viewer": {
+							DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+								typesystem.RelationReference("user", ""),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	err := ds.WriteAuthorizationModel(ctx, store, model)
+	require.NoError(b, err)
+
+	n := 0
+	for i := 0; i < 200; i++ {
+		var tuples []*openfgapb.TupleKey
+
+		for j := 0; j < 100; j++ {
```

Comment:
`< 100` could be `< ds.MaxTuplesInWriteOperation()`

But `i < 200` looks like an arbitrary number

Reply:
> 100 is the max concurrency limit

The call to `ds.Write(tuples)` will fail if you pass in more tuples that the data storage can accept

Reply:
Resolved